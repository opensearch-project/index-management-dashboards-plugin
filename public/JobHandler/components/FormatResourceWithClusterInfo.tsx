/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */
import React from "react";
import { ClusterInfo } from "../../models/interfaces";

export interface FormatResourceWithClusterInfoProps {
  clusterInfo?: ClusterInfo;
}

export const FormatResourceWithClusterInfo = (props: { resource: string } & FormatResourceWithClusterInfoProps) => {
  return props.clusterInfo?.cluster_name ? (
    <>
      {props.clusterInfo.cluster_name}/{props.resource}
    </>
  ) : (
    <>{props.resource}</>
  );
};

export const FormatResourcesWithClusterInfo = (props: { resources: string[] } & FormatResourceWithClusterInfoProps) => {
  return (
    <>
      {props.resources.map((item, index) => (
        <span key={item}>
          {index > 0 && ", "}
          <FormatResourceWithClusterInfo resource={item} clusterInfo={props.clusterInfo} />
        </span>
      ))}
    </>
  );
};
