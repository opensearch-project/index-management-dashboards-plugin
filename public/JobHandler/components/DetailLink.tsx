/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */
import React from "react";
import { EuiLink } from "@elastic/eui";
import { ROUTES } from "../../utils/constants";
import { FormatResourceWithClusterInfo, FormatResourceWithClusterInfoProps } from "./FormatResourceWithClusterInfo";

export const DetailLink = (props: FormatResourceWithClusterInfoProps & { index: string; writingIndex?: string }) => {
  return (
    <EuiLink href={`#${ROUTES.INDEX_DETAIL}/${props.writingIndex ? props.writingIndex : props.index}`}>
      <FormatResourceWithClusterInfo resource={props.index} clusterInfo={props.clusterInfo} />
    </EuiLink>
  );
};
