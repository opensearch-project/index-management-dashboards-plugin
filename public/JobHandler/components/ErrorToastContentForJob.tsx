/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */
import { EuiButton, EuiSpacer, EuiText } from "@elastic/eui";
import React from "react";
import { Modal } from "../../components/Modal";

export const ErrorToastContentForJob = (props: { shortError?: React.ReactChild; fullError?: React.ReactChild }) => {
  const { shortError = null, fullError } = props;
  return (
    <div>
      <div>{shortError}</div>
      {fullError ? (
        <>
          <EuiSpacer />
          <EuiButton
            onClick={
              /* istanbul ignore next */ () => {
                Modal.show({
                  locale: {
                    ok: "Close",
                  },
                  title: shortError || "",
                  content: (
                    <EuiText>
                      <div>{fullError}</div>
                    </EuiText>
                  ),
                });
              }
            }
            style={{ float: "right" }}
            color="danger"
          >
            See full error
          </EuiButton>
        </>
      ) : null}
    </div>
  );
};
