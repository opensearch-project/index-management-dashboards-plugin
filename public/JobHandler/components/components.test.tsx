/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */
import React from "react";
import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { DetailLink } from "./DetailLink";
import { ErrorToastContentForJob } from "./ErrorToastContentForJob";
import { FormatResourceWithClusterInfo, FormatResourcesWithClusterInfo } from "./FormatResourceWithClusterInfo";

describe("<DetailLink /> spec", () => {
  it("render with default props", async () => {
    const { container } = render(<DetailLink index="index" />);
    expect(container).toMatchSnapshot();
  });

  it("render with writing index", async () => {
    const { container } = render(<DetailLink index="index" writingIndex="writingIndex" />);
    expect(container).toMatchSnapshot();
  });

  it("render with cluster info", async () => {
    const { container } = render(
      <DetailLink
        index="index"
        writingIndex="writingIndex"
        clusterInfo={{
          cluster_name: "cluster_name",
        }}
      />
    );
    expect(container).toMatchSnapshot();
  });
});

describe("<ErrorToastContentForJob /> spec", () => {
  it("render with default props", async () => {
    const { container } = render(<ErrorToastContentForJob />);
    expect(container).toMatchSnapshot();
  });

  it("render with action", async () => {
    const { container, getByText, findByText } = render(<ErrorToastContentForJob shortError="shortError" fullError="fullError" />);
    expect(container).toMatchSnapshot();
    await userEvent.click(getByText("See full error"));
    await findByText("fullError");
  });
});

describe("<FormatResourceWithClusterInfo /> spec", () => {
  it("render with default props", async () => {
    const { container } = render(<FormatResourceWithClusterInfo resource="index" />);
    expect(container).toMatchSnapshot();
  });

  it("render with cluster info", async () => {
    const { container } = render(
      <FormatResourceWithClusterInfo
        resource="index"
        clusterInfo={{
          cluster_name: "cluster_name",
        }}
      />
    );
    expect(container).toMatchSnapshot();
  });
});

describe("<FormatResourcesWithClusterInfo /> spec", () => {
  it("render with default props", async () => {
    const { container } = render(<FormatResourcesWithClusterInfo resources={["index", "index1", "index2"]} />);
    expect(container).toMatchSnapshot();
  });

  it("render with cluster info", async () => {
    const { container } = render(
      <FormatResourcesWithClusterInfo
        resources={["index", "index1", "index2"]}
        clusterInfo={{
          cluster_name: "cluster_name",
        }}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
