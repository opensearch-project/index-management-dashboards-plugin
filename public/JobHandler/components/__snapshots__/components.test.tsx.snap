// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetailLink /> spec render with cluster info 1`] = `
<div>
  <a
    class="euiLink euiLink--primary"
    href="#/index-detail/writingIndex"
    rel="noreferrer"
  >
    cluster_name
    /
    index
  </a>
</div>
`;

exports[`<DetailLink /> spec render with default props 1`] = `
<div>
  <a
    class="euiLink euiLink--primary"
    href="#/index-detail/index"
    rel="noreferrer"
  >
    index
  </a>
</div>
`;

exports[`<DetailLink /> spec render with writing index 1`] = `
<div>
  <a
    class="euiLink euiLink--primary"
    href="#/index-detail/writingIndex"
    rel="noreferrer"
  >
    index
  </a>
</div>
`;

exports[`<ErrorToastContentForJob /> spec render with action 1`] = `
<div>
  <div>
    <div>
      shortError
    </div>
    <div
      class="euiSpacer euiSpacer--l"
    />
    <button
      class="euiButton euiButton--danger"
      style="float: right;"
      type="button"
    >
      <span
        class="euiButtonContent euiButton__content"
      >
        <span
          class="euiButton__text"
        >
          See full error
        </span>
      </span>
    </button>
  </div>
</div>
`;

exports[`<ErrorToastContentForJob /> spec render with default props 1`] = `
<div>
  <div>
    <div />
  </div>
</div>
`;

exports[`<FormatResourceWithClusterInfo /> spec render with cluster info 1`] = `
<div>
  cluster_name
  /
  index
</div>
`;

exports[`<FormatResourceWithClusterInfo /> spec render with default props 1`] = `
<div>
  index
</div>
`;

exports[`<FormatResourcesWithClusterInfo /> spec render with cluster info 1`] = `
<div>
  <span>
    cluster_name
    /
    index
  </span>
  <span>
    , 
    cluster_name
    /
    index1
  </span>
  <span>
    , 
    cluster_name
    /
    index2
  </span>
</div>
`;

exports[`<FormatResourcesWithClusterInfo /> spec render with default props 1`] = `
<div>
  <span>
    index
  </span>
  <span>
    , 
    index1
  </span>
  <span>
    , 
    index2
  </span>
</div>
`;
